addons/hr_holidays/models/hr_leave_accrual_plan_level.py

```diff
@@ -12,16 +12,26 @@
 
 DAYS = ['sun', 'mon', 'tue', 'wed', 'thu', 'fri', 'sat']
 MONTHS = ['jan', 'feb', 'mar', 'apr', 'may', 'jun', 'jul', 'aug', 'sep', 'oct', 'nov', 'dec']
+# Used for displaying the days and reversing selection -> integer
+DAY_SELECT_VALUES = [str(i) for i in range(1, 29)] + ['last']
+# Integer to ordinal string, found of stackoverflow: https://stackoverflow.com/a/50992575
+DAY_SELECT_DISPLAYS = ["%d%s" % (i, "tsnrhtdd" [(i / 10 % 10 != 1) * (i % 10 < 4) * i % 10::4]) for i in range(1, 29)] + ['last day']
+DAY_SELECT = [val for val in zip(DAY_SELECT_VALUES, DAY_SELECT_DISPLAYS)]
 
 def _get_date_check_month(year, month, day):
     """
-    Returns the day it would be if day is outside of the month's range
-    Example: 2021 feb 30 -> 2021 mar 2
+    Returns the date given or the last day of the month if day is equal to 29
```

Comment:
Side note: not my module and there are probably a lot of things I don't know about what you do, but are you sure you cannot use standard date management tools available in standard python libraries (+ maybe babel for formatting ) ?

Reply:
I am not knowledgable enough on the python libraries to know if there is a standard way to do this, it might be good to take the time to check and come back on it, but I doubt there is, since we try to apply regular maths to dates which are not made for it

Reply:
I am not sure to understand all the scenarios but why this couldn't be achieved with `dateutil.relativedelta` ?

Reply:
I don't have my whole mind into it right now but basically the user can select last day of the month, which wouldn't work with something like `relativedelta(months=1)`, I am not saying this solution is optimal but it works for now and could be cleaned up later :)

Reply:
Oh nice I hadn't thought about that, I will try it out, thanks you very much

Reply:
Indeed I made it work with standard stuff, thanks a lot for the tips